server.js

```diff
@@ -0,0 +1,306 @@
+'use strict';
+
+const http = require('http');
+const fs = require('fs');
+const qs = require('querystring');
+const adminPrivilege = {
+  brett: 'password',
+};
+
+const server = http.createServer((req, res) => {
+  if (req.method === 'GET') {
+    if (req.url === '/') {
+      fs.readFile('./public/index.html', 'utf8', (err, data) => {
+        if (err) {
+          throw err;
+        }
+
+        return res.end(data);
+      });
+    }
+
+    if (req.url === '/helium.html') {
+      fs.readFile('./public/helium.html', 'utf8', (err, data) => {
+        if (err) {
+          throw err;
+        }
+
+        return res.end(data);
+      });
+    }
+
+    if (req.url === '/hydrogen.html') {
+      fs.readFile('./public/hydrogen.html', 'utf8', (err, data) => {
+        if (err) {
+          throw err;
+        }
+
+        return res.end(data);
+      });
+    }
+
+    if (req.url === '/css/styles.css') {
+      fs.readFile('./public/css/styles.css', 'utf8', (err, data) => {
+        if (err) {
+          throw err;
+        }
+
+        return res.end(data);
+      });
+    } else {
+      fs.readFile('./public/404.html', 'utf8', (err, data) => {
+        if (err) {
+          throw err;
+        }
+
+        return res.end(data);
+      });
+    }
+  }
+
+  if (req.method === 'POST') {
+    if (req.url === '/elements') {
+      if (req.headers.authorization) {
+        var encodedString = req.headers.authorization.slice(6);
+        var base64Buffer = new Buffer(encodedString, 'base64');
+        var decodedString = base64Buffer.toString();
+        let username = decodedString.slice(0, decodedString.indexOf(':'));
+        let password = decodedString.slice(decodedString.indexOf(':') + 1);
+
+        if (adminPrivilege.hasOwnProperty(username) && adminPrivilege[username] === password) {
+          let totalData = '';
+
+          req.on('data', (data) => {
+            totalData += data;
+          });
+
+          req.on('end', () => {
+            const parsedData = qs.parse(totalData);
+
+            const template = `<!DOCTYPE html>
+            <html lang="en">
+            <head>
+            <meta charset="UTF-8">
+            <meta name="viewport" content="width=device-width, initial-scale=1.0">
+            <meta http-equiv="X-UA-Compatible" content="ie=edge">
+            <title>${parsedData.elementName}</title>
+            </head>
+            <body>
+            <div>${parsedData.elementName}</div>
+            <div>${parsedData.elementSymbol}</div>
+      <div>${parsedData.elementAtomicNumber}</div>
+      <div>${parsedData.elementDescription}</div>
+      </body>
+      </html>`;
+
+            fs.writeFile(`./public/${parsedData.elementName}.html`, template, 'utf8', (err) => {
+              if (err) {
+                throw err;
+              }
+            });
+
+            fs.readdir('./public', (err, files) => {
```

Comment:
Your fs.readdir should be inside the callback of your fs.writeFile.

Reply:
You don't handle the err that comes back from fs.readdir. This should be sending a 500 response back to the client.